zerver/data_import/rocketchat.py

```diff
@@ -0,0 +1,675 @@
+    subscriber_handler: SubscriberHandler,
+) -> None:
+    stream_members_map: Dict[int, Set[int]] = {}
+
+    for rc_user_id in user_id_to_user_map:
+        user_dict = user_id_to_user_map[rc_user_id]
+
+        if "__rooms" not in user_dict:
+            continue
+
+        for channel in user_dict["__rooms"]:
+            if channel in dsc_id_to_dsc_map:
+                # Ignore discussion rooms
+                continue
+            stream_id = stream_id_mapper.get(channel)
+            if stream_id not in stream_members_map:
+                stream_members_map[stream_id] = set()
+            stream_members_map[stream_id].add(user_id_mapper.get(rc_user_id))
+
+    for stream in zerver_stream:
+        if stream["id"] in stream_members_map:
+            users = stream_members_map[stream["id"]]
+        else:
+            users = set()
+        subscriber_handler.set_info(users=users, stream_id=stream["id"])
+
+
+def build_reactions(
+    total_reactions: List[ZerverFieldsT],
+    reactions: List[Dict[str, Any]],
+    message_id: int,
+    # zerver_realmemoji: List[ZerverFieldsT],
+) -> None:
+    # For the Unicode emoji codes, we use equivalent of
+    # function 'emoji_name_to_emoji_code' in 'zerver/lib/emoji' here
+    for reaction in reactions:
+        emoji_name = reaction["name"]
+        user_id = reaction["user_id"]
+        # Check in Unicode emoji
+        if emoji_name in name_to_codepoint:
+            emoji_code = name_to_codepoint[emoji_name]
+            reaction_type = Reaction.UNICODE_EMOJI
+        else:  # nocoverage
+            continue
+
+        reaction_id = NEXT_ID("reaction")
+        reaction = Reaction(
+            id=reaction_id,
+            emoji_code=emoji_code,
+            emoji_name=emoji_name,
+            reaction_type=reaction_type,
+        )
+
+        reaction_dict = model_to_dict(reaction, exclude=["message", "user_profile"])
+        reaction_dict["message"] = message_id
+        reaction_dict["user_profile"] = user_id
+        total_reactions.append(reaction_dict)
+
+
+def process_raw_message_batch(
+    realm_id: int,
+    raw_messages: List[Dict[str, Any]],
+    subscriber_map: Dict[int, Set[int]],
+    user_handler: UserHandler,
+    is_pm_data: bool,
+    output_dir: str,
+    # zerver_realmemoji: List[Dict[str, Any]],
+    total_reactions: List[ZerverFieldsT],
+) -> None:
+    def fix_mentions(content: str, mention_user_ids: Set[int]) -> str:
+        for user_id in mention_user_ids:
+            user = user_handler.get_user(user_id=user_id)
+            rc_mention = "@{short_name}".format(**user)
+            zulip_mention = "@**{full_name}**".format(**user)
+            content = content.replace(rc_mention, zulip_mention)
+
+        content = content.replace("@all", "@**all**")
+        # We don't have an equivalent for Mattermost's @here mention which mentions all users
+        # active in the channel.
+        content = content.replace("@here", "@**all**")
```

Comment:
This comment is kinda confusing; we should just support exactly the set of wildcard mentions that exist in Rocket.Chat, without reference to what Mattermost happens to do.

Reply:
I actually copied the comment from the Mattermost tool, which is " We don't have an equivalent for Mattermost's @here mention which mentions all users online in the channel.". I changed the `online` to `active`, as RocketChat tags all the active users in a channel with `@here`, but forgot to change the name.